File: 435-non-overlapping-intervals/non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int> &a, vector<int> &b) -> bool {
            if(a[1] != b[1]) return a[1] < b[1];
            else return a[0] > b[0];
        });
        int end = intervals[0][1];
        int count = 0;
        for(int i=1; i<intervals.size(); i++){
            if(intervals[i][0] >= end){
                end = intervals[i][1];
            } else count++;
        }
        return count;
    }
};

File: 435-non-overlapping-intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: [x[1], x[0]])
        # print(intervals)

        if len(intervals) == 0:
            return []
        
        start = intervals[0][0]
        end = intervals[0][1]
        idx = 1
        count = 0
        while(idx < len(intervals)):
            if end > intervals[idx][0]:
                # print(end, intervals[idx][0])
                count += 1
            else:
                start = intervals[idx][0]
                end = intervals[idx][1]
            
            idx += 1
        
        return count
            
            
        